src/Microsoft.SqlTools.ServiceLayer/QueryExecution/Query.cs

```diff
@@ -374,6 +472,14 @@ private void OnInfoMessage(object sender, SqlInfoMessageEventArgs args)
             }
         }
 
+        /// <summary>
+        /// Function to add a new batch to a Batch set
+        /// </summary>
+        private void addBatch(string query, List<Batch> batchSet, IFileStreamFactory outputFactory)
+        {
+            batchSet.Add(new Batch(query, new SelectionData(0, 0, 0, 0), batchSet.Count, outputFactory));
```

Comment:
Ideally, I think if we can send null for the selectiondata we might be better off. I'm thinking how this will play out for our ability to execute arbitrary queries.